server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -1679,4 +1681,16 @@ static void validateTranslogDurabilitySettings(Settings requestSettings, Cluster
         }
 
     }
+
+    public static void validateIndexStoreLocality(Settings indexSettings) {
```

Comment:
Don't see a test for this case where we try creating an index with INDEX_STORE_LOCALITY_SETTING and when feature flag is disabled.